arangod/Aql/Expression.cpp

```diff
@@ -382,91 +499,6 @@ void Expression::prepareForExecution(ExpressionContext& ctx) {
   }
 }
 
-// brief execute an expression of type SIMPLE, the convention is that
-// the resulting AqlValue will be destroyed outside eventually
-AqlValue Expression::executeSimpleExpression(ExpressionContext& ctx,
-                                             AstNode const* node,
-                                             bool& mustDestroy, bool doCopy) {
-  switch (node->type) {
```

Comment:
Btw can't compiler generate same code for this switch?
I thought switch everything better than manually written map.
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78113
(all variant implementation using switch for visiting now (they use table when types count more than some number, [for msvc 256](https://github.com/microsoft/STL/blob/main/stl/inc/variant#L721)))

Reply:
I was also disappointed that this change improved performance at least on the instances we tried it on.